### DISCUSSION TITLE
Open source and free software is good, but...

### DISCUSSION THREAD

**Discussion by EduardoVB** [twinbasic/twinbasic#700] [Ideas]
Lately I've been considering how the software world evolved.
VB6 in the first times developed into an ecosystem of mostly closed-source proprietary and commercial components. Some companies appeared selling components and also independent developers did.

Then, slowly, things started to change.
On one hand it was Stallman evangelizing with his ideas that the developers just had to get paid for assistance, not for the software itself.
IMO (you could not share it) he made lot of harm with those ideas.
Many developers bought the idea of working for free. One could think difficult to do, but happened.

There was another trend that can be seen as a progress, but at the end it probably also just damaged developers: StackOverflow. It seems that now, if you are an expert and spent 20 years learning, it worth nothing. 
Before you could sell your services, your expertise as a consultant (and it wasn't cheap, as our field of expertise is not easy).

Big companies keep profiting (and more than ever), they always find the way, they are not affected by these issues. What Stallman did only had an impact in independent developers and small companies.

I don't know of any other field where the knowledge worth nothing, and the work either.
Nobody can change the mentality of every developer now, the damage is done.

But I was thinking that perhaps tB could have some way, I don't mean for all all, because having basic libraries for free and open source is good, but if we could "invent" some way where the developers are paid for their works, if they choose to do so, it could be good for the ecosystem. Open source or not, we will have to think about that too.

I don't know how, I just "know" that the work should be paid, and I also guess that if developers are paid, the ecosystem around it will develop much larger and probably with better quality.

What do you think?

**Comment by WaynePhillipsEA** [comment]
I do generally like this idea, but like you, I'm unsure how it would work in practice.  

Once tB supports *compiled* packages, I guess(?) some developers would certainly like to offer commercial packages.

I had a crazy idea of possibly doing crowdfunding to somehow encourage developers to participate in creating (quality) tB packages, with the money raised being split between package creators, on say a monthly basis.  But I think it would get complicated quickly... so that's probably a non-starter.  

I'd be keen to hear others' views and ideas around this subject.

**Reply by Kr00l** [reply]
Ideally it would work like a store. Where each developer get's paid via the store for each sell and tB withholds a certain fee for itself.

But that's a huuuge thing and I guess should be post v7 or so. :)

**Reply by uwbwsvd** [reply]
For me is the question. Should tB be for mostly hobbiest or professional. Office with ms access costs approx 6 euros more per month with office. If I need MS Access because its the tool the fits my needs, i will pay for it. So I would do with tB. If I get work done easier, faster, better, I would pay a price no hobbiest would think about to pay.

**Reply by mansellan** [reply]
I think twinBASIC should support both FOSS (permissive or copyleft) and commercial (ideally with time-limited trials).

Compiled packages will be key for the latter. I'm guessing that compiled twinBASIC is just as resilient to decompilation as C. It would be nice if there were a web store to browse packages, and showcase new and trending ones like NuGet does.

FOSS has definitely been a complete game changer in the past couple of decades, but commercial is still alive and well. One needs only look at the high-profile issues (OpenSSL) caused by thousands of companies relying on the work of volunteers without giving back.

https://xkcd.com/2347/
